### PR TITLE
Add is_ok?/1 and is_error?/1 helper functions

### LINKED ISSUE #59: Proposal: add `ok?` and `error?` methods
Could be cool to have some other helpers to check if result tuple is in error or ok.

The use case in my mind is filter a collection of result tuples looking only for ok results.

Thank you in advance

### PR DESCRIPTION
**Pull request by thomasnaudin**
Fix #59

### CODE REVIEW

File: lib/ok.ex [hunk 1]
```diff
@@ -116,6 +116,40 @@ defmodule OK do
 
   def check({:error, reason}, _func, _reason), do: {:error, reason}
 
+  @doc """
+  Helper function. Checks if a result tuple is tagged as `:ok`, and returns `true` if so.
```
Comment (CrowdHailer) [inline lib/ok.ex]: Don't bother with `Helper function` (my opinion is they're all helpful) :-)

Comment (thomasnaudin) [reply lib/ok.ex]: Done ;)

**Comment by thomasnaudin** [comment]
I was thinking about harmonizing semantic.
There's already `success/1` and `failure/1` functions, maybe better names for those new ones would be `is_success?/1` and `is_failure?/1` ?

**Comment by CrowdHailer** [comment]
It's a good question. There are a lot of OK users I wish I could ask them but I don't know anywhere I would find answers.

**Comment by CrowdHailer** [comment]
I don't have too strong a feeling on this but I think my inclination is towards success/failure

**Comment by thomasnaudin** [comment]
I agree – I updated the PR.

**Comment by CrowdHailer** [comment]
Released in 2.2 https://hex.pm/packages/ok/2.2.0

**Pull request merged**